File: SuitePlugins/MapPacker/src/main/java/MapPlugin.java
import store.plugin.Plugin;
import suite.annotation.PluginDescriptor;
import store.plugin.PluginType;

/**
 * @author Andrew on 11/25/2019
 * @project ValkyrCacheSuite
 */
@PluginDescriptor(author = "ReverendDread", description = "Pack's maps into the cache.", version = "183", type = PluginType.MAP)
public class MapPlugin extends Plugin {

    public static UnderlayLoader underlays;
    public static OverlayLoader overlays;

    @Override
    public boolean load() {
        setController(new MapPacker());
        underlays = new UnderlayLoader().load();
        overlays = new OverlayLoader().load();
        return true;
    }

    @Override
    public String getFXML() {
        return "main.fxml";
    }
}

File: SuitePlugins/MapPacker/src/main/kotlin/dev/dread/map/MapPlugin.kt
package dev.dread.map

import dev.dread.map.loader.OverlayLoader
import dev.dread.map.loader.UnderlayLoader
import dev.dread.map.packer.MapPacker
import store.plugin.Plugin
import store.plugin.PluginType
import suite.annotation.PluginDescriptor

@PluginDescriptor(author = "ReverendDread", description = "Pack's maps into the cache.", version = "183", type = PluginType.MAP)
class MapPlugin : Plugin() {

    companion object {
        internal lateinit var underlays: UnderlayLoader
        internal lateinit var overlays: OverlayLoader
    }

    override fun load(): Boolean {
        controller = MapPacker()
        underlays = UnderlayLoader().load()
        overlays = OverlayLoader().load()
        return true
    }

    override fun getFXML(): String = "main.fxml"

}